pytest_examples/test_class.py

```diff
@@ -29,3 +29,72 @@ def test_add_floats(self):
     def test_add_floats_and_int(self):
         instance = ClassToTest()
         assert instance.add(1, 2.1) == pytest.approx(3.1)
+
+
+class TestMultiple:
+    def test_multiple_string(self):
+        instance = ClassToTest()
+        with pytest.raises(TypeError):  
+            instance.multiply("2", "3")  
+
+    def test_multiple_arrays(self):
+        instance = ClassToTest()
+        with pytest.raises(TypeError): 
+            instance.multiply([3, 4, 5], 1)
+
+    def test_multiple_ints(self):
+        instance = ClassToTest()
+        assert instance.multiply(2, 3) == 6
+        assert instance.multiply(3, 3) == 9
+        assert instance.multiply(6, 2) == 12
+
+    def test_multiple_floats(self):
+        instance = ClassToTest()
+        instance.multiply(2.1, 3.2) 
+        assert instance.multiply(1.2, 2.1) == pytest.approx(2.52)
+        assert instance.multiply(-1.1, 2.1) == pytest.approx(-2.31)
+        assert instance.multiply(0.0, 1.5) == pytest.approx(0)
+
+    def test_multiple_floats_and_int(self): 
+        instance = ClassToTest()
+        assert instance.multiply(1, 2.1) == pytest.approx(2.1)      
+
+
+
+class TestDivide:
+    def test_divide_string(self):
+        instance = ClassToTest()
+        with pytest.raises(TypeError):
+            instance.divide("5", "6")
+
+    def test_divide_int(self):
+        instance = ClassToTest()
+        with pytest.raises(TypeError):
+            instance.divide(5, 6)
+
+    def test_divide_array(self):
+        instance = ClassToTest()
+        with pytest.raises(TypeError):
+            instance.divide([1, 6, 7], [6, 7, 8])
+
+    def test_divide_int(self):
+        instance = ClassToTest()
+        assert instance.divide(4, 2) == 2
+        assert instance.divide(10, 2) == 5
+        assert instance.divide(6, 2) == 3
```

Comment:
I think you are missing a pretty important case in the divide test. can you guess what it is?

Reply:
Add a test case for if you divide by 0, an error is thrown.